recipes/flatcc/all/conanfile.py

```diff
@@ -0,0 +1,107 @@
+                "fast_double": [True, False],
+                "ignore_const_condition": [True, False],
+    }
+    default_options = { "shared": True,
+                        "fPIC": True,
+                        "portable": False,
+                        "gnu_posix_memalign": True,
+                        "runtime_lib_only": False,
+                        "verify_assert": False,
+                        "verify_trace": False,
+                        "reflection": True,
+                        "native_optim": False,
+                        "fast_double": False,
+                        "ignore_const_condition": False
+    }
+    settings = "os", "arch", "compiler", "build_type"
+    no_copy_source = True
+
+    _cmake = None
+
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def configure(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+        if self.settings.compiler == "Visual Studio" and self.options.shared:
+            #Building flatcc shared libs with Visual Studio is broken
+            raise ConanInvalidConfiguration("Building flatcc libraries shared is not supported")
+        if self.settings.os == "Windows" and self.settings.compiler == "gcc":
+            raise ConanInvalidConfiguration("Building flatcc with MinGW is not supported")
+        if self.settings.os == "Macos" and self.options.shared:
+            raise ConanInvalidConfiguration("Building flatcc shared on Macos is currently not supported due to Systems Integrity Protection problems")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def _configure_cmake(self):
+        if not self._cmake:
+            self._cmake = CMake(self)
+            self._cmake.definitions["FLATCC_PORTABLE"] = self.options.portable
+            self._cmake.definitions["FLATCC_GNU_POSIX_MEMALIGN"] = self.options.gnu_posix_memalign
+            self._cmake.definitions["FLATCC_RTONLY"] = self.options.runtime_lib_only
+            self._cmake.definitions["FLATCC_INSTALL"] = True
+            self._cmake.definitions["FLATCC_COVERAGE"] = False
+            self._cmake.definitions["FLATCC_DEBUG_VERIFY"] = self.options.verify_assert
+            self._cmake.definitions["FLATCC_TRACE_VERIFY"] = self.options.verify_trace
+            self._cmake.definitions["FLATCC_REFLECTION"] = self.options.reflection
+            self._cmake.definitions["FLATCC_NATIVE_OPTIM"] = self.options.native_optim
+            self._cmake.definitions["FLATCC_FAST_DOUBLE"] = self.options.fast_double
+            self._cmake.definitions["FLATCC_IGNORE_CONST_COND"] = self.options.ignore_const_condition
+            self._cmake.definitions["FLATCC_TEST"] = False
+            self._cmake.configure(source_folder=os.path.join(self.source_folder, self._source_subfolder))
```

Comment:
By running the cmake script from the source_subfolder directly,
you're not executing
```
include(conanbuildinfo.cmake)
conan_basic_setup()
```

Which causes the compiler to use the wrong runtime. This is a problem with MSVC.

Reply:
Thanks for (again) saving me a lot of analysis time. Strangely the current recipe builds fine on my Windows 10 VM with Visual Studio 2019.
I will try to fix the Windows build later this week, some other priorities came up.

Reply:
The problem is always with `build_Type=Debug` and `runtime=MTd` or `MDd`.
Real troublemakers, those runtimes are... :smile: